Content.Shared/Projectiles/HasProjectilesEmbeddedComponent.cs

```diff
@@ -0,0 +1,23 @@
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Xml;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Projectiles;
+
+/// <summary>
+/// This component marks that its entity has projectiles embedded in it, and tracks what those projectile entities are.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class HasProjectilesEmbeddedComponent : Component
+{
+    [DataField, AutoNetworkedField, ViewVariables(VVAccess.ReadOnly)]
+
+    private List<EntityUid> _embeddedProjectiles = [];
+
+    public void Add(Entity<EmbeddableProjectileComponent> projectile) => _embeddedProjectiles.Add(projectile);
+    public bool Contains(Entity<EmbeddableProjectileComponent> projectile) => _embeddedProjectiles.Contains(projectile);
+    public void Remove(Entity<EmbeddableProjectileComponent> projectile) => _embeddedProjectiles.Remove(projectile);
+    public bool IsEmpty() => _embeddedProjectiles.Count == 0;
```

Comment:
I added these to enforce that the projectile has `EmbeddableProjectileComponent` via `Entity<EmbeddableProjectileComponent>`.
Lemme know if this is undesirable and I can just inline these functions.

Reply:
I don't personally think this adds much. There's no guarantee that the entities in the list will continue to have that component, and if the component does get removed for whatever reason, you'll no longer be able to call this `Remove` to remove it from this list. (Also, `Remove` should return a boolean.)

Reply:
> There's no guarantee that the entities in the list will continue to have that component
This seems like it kinda undermines the purpose of `Entity<TComp>`, but I guess in my case, holding onto entities is the real culprit of that possible misalignment.

Good callout.

Reply:
> Also, Remove should return a boolean

Coming from lots of experience in Kotlin where
```koitlin
fun someFun(): Unit = returnsBool()
```
would complain, I blame C# :P